sdk/trace/id_generator_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	// Can't assert values of IDs but can assert they're valid, try a lot as a sort of fuzz check.
+	_tryTimes    = 1000
+	_testTraceID = "4bf92f3577b34da6a3ce929d0e0e4736"
+)
+
+func TestNewIDs(t *testing.T) {
+	gen := defaultIDGenerator()
+
+	for i := 0; i < _tryTimes; i++ {
+		traceID, spanID := gen.NewIDs(context.Background())
+		assert.True(t, traceID.IsValid(), fmt.Sprintf("trace id: %s", traceID.String()))
+		assert.True(t, spanID.IsValid(), fmt.Sprintf("span id: %s", spanID.String()))
+	}
+}
+
+func TestNewSpanID(t *testing.T) {
+	gen := defaultIDGenerator()
+	traceId, err := trace.TraceIDFromHex(_testTraceID)
```

Comment:
Can we use a "hardcoded" `TraceID` (type) value instead of using a `string`+`trace.TraceIDFromHex`

This test may stop working if `trace.TraceIDFromHex` has a bug

Reply:
That's a good point. Updated